network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractTopology.java

```diff
@@ -358,18 +362,76 @@ private void setCalculatedBuses(Resource<VoltageLevelAttributes> voltageLevelRes
+                    }
+                });
+            if (foundConfiguredBus.get() != null) {
+                v.set(foundConfiguredBus.get().getResource().getAttributes().getV());
+                angle.set(foundConfiguredBus.get().getResource().getAttributes().getAngle());
+            }
+        });
+    }
+
+    private void getVAndAngleFromOtherView(NetworkObjectIndex index,
+                                           Resource<VoltageLevelAttributes> voltageLevelResource,
+                                           ConnectedSetResult<T> connectedSet,
+                                           AtomicDouble v,
+                                           AtomicDouble angle,
+                                           boolean isBusView) {
+        AtomicBoolean foundInCalculatedBuses = new AtomicBoolean(false);
+        List<CalculatedBusAttributes> calculatedBusAttributes = isBusView ?
+            voltageLevelResource.getAttributes().getCalculatedBusesForBusBreakerView() :
+            voltageLevelResource.getAttributes().getCalculatedBusesForBusView();
+        if (!CollectionUtils.isEmpty(calculatedBusAttributes)) {
+            connectedSet.getConnectedVertices().forEach(vertex ->
+                calculatedBusAttributes.stream().filter(attr -> attr.getVertices().contains(vertex)).forEach(b -> {
+                    if (!Double.isNaN(b.getV())) {
+                        v.set(b.getV());
+                        foundInCalculatedBuses.set(true);
+                    }
+                    if (!Double.isNaN(b.getAngle())) {
+                        angle.set(b.getAngle());
+                        foundInCalculatedBuses.set(true);
+                    }
+                }));
+        }
+        if (isBusView && !foundInCalculatedBuses.get()) {
+            // get V and Angle values from configured buses
+            getVAndAngleFromConfiguredBus(index, voltageLevelResource, connectedSet, v, angle);
+        }
+    }
+
     private CalculationResult<T> getCalculatedBusAttributesList(NetworkObjectIndex index, Resource<VoltageLevelAttributes> voltageLevelResource, boolean isBusView) {
         List<CalculatedBusAttributes> calculatedBusAttributesList;
         Map<T, Integer> nodeOrBusToCalculatedBusNum;
         if (isCalculatedBusesValid(voltageLevelResource, isBusView)) {
             calculatedBusAttributesList = isBusView ? voltageLevelResource.getAttributes().getCalculatedBusesForBusView() : voltageLevelResource.getAttributes().getCalculatedBusesForBusBreakerView();
             nodeOrBusToCalculatedBusNum = getNodeOrBusToCalculatedBusNum(voltageLevelResource, isBusView);
         } else {
-            // calculate buses
             List<ConnectedSetResult<T>> connectedSetList = findConnectedSetList(index, voltageLevelResource, isBusView);
+            AtomicDouble v = new AtomicDouble(Double.NaN);
+            AtomicDouble angle = new AtomicDouble(Double.NaN);
```

Comment:
v set in all remaining buses

Reply:
Ah I see this bug made the AbstractMainConnectedComponentWithSwitchTest work because there is a bus without any vertex which would be unset, but since it's setting v in all remaining buses it sets it anyway.

This is all to do with the fact that we can not use the equipment Vertex (=non switch equipment) to match buses, we need to use the topology vertex (nodes/configuredbus)